resources/services/wafregional/rate_based_rules.go

```diff
@@ -0,0 +1,178 @@
+				Description: "This is AWS WAF Classic documentation",
+				Resolver:    fetchWafregionalRateBasedRuleMatchPredicates,
+				Columns: []schema.Column{
+					{
+						Name:        "rate_based_rule_cq_id",
+						Description: "Unique CloudQuery ID of aws_wafregional_rate_based_rules table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "data_id",
+						Description: "A unique identifier for a predicate in a Rule, such as ByteMatchSetId or IPSetId",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "negated",
+						Description: "Set Negated to False if you want AWS WAF to allow, block, or count requests based on the settings in the specified ByteMatchSet, IPSet, SqlInjectionMatchSet, XssMatchSet, RegexMatchSet, GeoMatchSet, or SizeConstraintSet",
+						Type:        schema.TypeBool,
+					},
+					{
+						Name:        "type",
+						Description: "The type of predicate in a Rule, such as ByteMatch or IPSet.  This member is required.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchWafregionalRateBasedRules(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().WafRegional
+	var params wafregional.ListRateBasedRulesInput
+	for {
+		result, err := svc.ListRateBasedRules(ctx, &params, func(o *wafregional.Options) { o.Region = cl.Region })
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, item := range result.Rules {
+			detail, err := svc.GetRateBasedRule(
+				ctx,
+				&wafregional.GetRateBasedRuleInput{RuleId: item.RuleId},
+				func(o *wafregional.Options) { o.Region = cl.Region },
+			)
+			if err != nil {
+				return diag.WrapError(err)
+			}
+			if detail.Rule == nil {
+				continue
+			}
+			res <- *detail.Rule
+		}
+		if aws.ToString(result.NextMarker) == "" {
+			break
+		}
+		params.NextMarker = result.NextMarker
+	}
+	return nil
+}
+func postResolveTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().WafRegional
+	var arn string
+	arnValue := resource.Get("arn")
+	if v, ok := arnValue.(string); ok {
+		arn = v
+	} else {
+		arn = *arnValue.(*string)
+	}
+	params := wafregional.ListTagsForResourceInput{ResourceARN: &arn}
+	tags := make(map[string]string)
+	for {
+		result, err := svc.ListTagsForResource(ctx, &params)
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		if result.TagInfoForResource != nil {
+			for _, t := range result.TagInfoForResource.TagList {
```

Comment:
Aside from the post resolver issue, you can use `client.TagsIntoMap(result.TagInfoForResource.TagList, tags)` instead of this loop.

Reply:
I'll try to do that, thanks for the suggestion!

Reply:
Fixed.